contrib_bots/lib/gcalendar.py

```diff
@@ -0,0 +1,367 @@
+from __future__ import print_function
+import httplib2
+import logging
+import os
+import re
+import sys
+
+# Uses Google's Client Library
+#   pip install --upgrade google-api-python-client
+from apiclient import discovery, errors
+from oauth2client import client, tools
+from oauth2client.client import HttpAccessTokenRefreshError
+from oauth2client.file import Storage
+from oauth2client.tools import argparser
+
+# pip install --upgrade dateparser
+import dateparser
+
+from pytz.exceptions import UnknownTimeZoneError
+from tzlocal import get_localzone
+
+SCOPES = 'https://www.googleapis.com/auth/calendar'
+CLIENT_SECRET_FILE = '.gcal_client_secret.json'
+APPLICATION_NAME = 'Zulip Google Calendar Bot'
+
+class UnknownMessageTypeError(Exception):
+    def __init__(self, msgtype):
+        self.type = msgtype
+        self.message = 'Unknown message type "' + msgtype + '"'
+
+    def __str__(self):
+        return self.message
+
+class MessageParseError(Exception):
+    def __init__(self, eid):
+        self.errid = eid
+
+        details = ''
+
+        if eid is 1:
+            details = 'Unknown message format'
+        elif eid is 2:
+            details = 'Unknown date format'
+        elif eid is 3:
+            details = 'The specified timezone doesn\'t exist'
+
+        self.message = details
+
+    def __str__(self):
+        return self.message
+
+# NOTE ABOUT GOOGLE'S OAUTH
+# Right now there are some duplicates of this function in Zulip.
```

Comment:
I've rephrased it as `some duplicates of the get_credentials() function`, for the moment.

Let me know if this works for you :)